components/net/fetch/methods.rs

```diff
@@ -502,22 +508,31 @@ pub fn main_fetch(
     response
 }
 
-fn wait_for_response(response: &mut Response, target: Target, done_chan: &mut DoneChannel) {
+async fn wait_for_response(
```

Comment:
This function concerns me - looping on a try_recv is often bad for performance, and it's not clear to me how much the yield_now actually improves things. I propose changing DoneChannel to use tokio channels instead of crossbeam ones, which should avoid the issues with blocking that the previous implementation encountered.

Reply:
Gave this a shot, however there are some peculiarities with adopting tokio channels.
1.) These types are from a very old version of tokio (0.1), since we need compatibility with the older version of hyper `wrap_stream`.
2.) This required the use of `futures_compat` to wrap the Receiver with a compatibility shim, which takes ownership of the receiver object.
3.) The Receiver type doesn't support multiple consumers (no clone method), so we need to use a Mutex and swap out the object in memory when consuming the stream. This can only be done once and will panic if done again.